src/test/java/com/exasol/matcher/ResultSetMatcher.java

```diff
@@ -0,0 +1,119 @@
+package com.exasol.matcher;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Types;
+
+import org.hamcrest.Description;
+import org.hamcrest.TypeSafeMatcher;
+
+/**
+ * This matcher compares two result sets.
+ */
+public final class ResultSetMatcher extends TypeSafeMatcher<ResultSet> {
+    private final ResultSet expectedResultSet;
+
+    /**
+     * Creates a new instance of {@link ResultSetMatcher}.
+     * 
+     * @param expectedResultSet expected Result Set
+     */
+    public ResultSetMatcher(final ResultSet expectedResultSet) {
+        this.expectedResultSet = expectedResultSet;
+    }
+
+    /**
+     * Compares against a result set.
+     * 
+     * @param expectedResultSet expected result set
+     * @return an instance of {@link ResultSetMatcher}
+     */
+    public static ResultSetMatcher matchesResultSet(final ResultSet expectedResultSet) {
+        return new ResultSetMatcher(expectedResultSet);
+    }
+
+    @Override
+    protected boolean matchesSafely(final ResultSet actualResultSet) {
+        try {
+            return assertEqualResultSets(actualResultSet);
+        } catch (final SQLException exception) {
+            throw new AssertionError(
+                    "Assertion failed due to an unexpected SQL exception. Cause: " + exception.getSQLState(),
+                    exception);
+        }
+    }
+
+    @Override
+    public void describeTo(final Description description) {
+        description.appendValue(this.expectedResultSet);
+    }
+
+    private boolean assertEqualResultSets(final ResultSet actualResultSet) throws SQLException {
+        final int expectedColumnCount = this.expectedResultSet.getMetaData().getColumnCount();
+        final int actualColumnCount = actualResultSet.getMetaData().getColumnCount();
+        if (expectedColumnCount != actualColumnCount) {
+            return false;
+        }
+        boolean expectedNext;
+        do {
+            expectedNext = this.expectedResultSet.next();
+            if (expectedNext != actualResultSet.next()) {
+                return false;
+            }
+            if (this.expectedResultSet.isLast() != actualResultSet.isLast()) {
+                return false;
+            }
+            if (expectedNext) {
+                if (!rowMatches(actualResultSet, expectedColumnCount)) {
+                    return false;
+                }
+            }
+        } while (expectedNext);
+        return true;
+    }
+
+    private boolean rowMatches(ResultSet actualResultSet, int expectedColumnCount) throws SQLException {
```

Comment:
Rename method to `doesRowMatch()`
Rename first parameter to `actualRow`.
Make all parameters final.

Reply:
done